### PR TITLE
Live reload

### PR DESCRIPTION
**Pull request by sidharthv96**
## :bookmark_tabs: Summary

Adds live reload to the esbuild development server. Will make the lives of devs easier :)
Uses Server Sent Events

Resolves #<your issue id here>

## :straight_ruler: Design Decisions

Could not use esbuild's watch as the rebuild events were not available easily. 
Chokidar is already used to watch docs, so not an extra dependency.

### :clipboard: Tasks

Make sure you

- [ ] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [ ] :computer: have added necessary unit/e2e tests.
- [ ] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/community/development.md#3-update-documentation) is used for all new features.
- [ ] :bookmark: targeted `develop` branch

### CODE REVIEW

File: demos/dev/example.html [hunk 1]
```diff
@@ -30,5 +30,32 @@
...
+          const time = JSON.parse(event.data);
+          if (time && time > loadTime) {
+            location.reload();
+          }
+        };
+        events.onerror = () => {
+          events.close();
+          // Try to reconnect after 1 second in case of errors
+          setTimeout(connectToEvents, 1000);
+        };
```
Comment (Yokozuna59) [inline demos/dev/example.html:51]: Maybe we should also log error?

File: .esbuild/server.ts
Comment (Yokozuna59) [inline .esbuild/server.ts:1]: Should we create types here or we can ignore it?

**Review by Yokozuna59 — COMMENTED** [review]
Some questions.

File: package.json [hunk 2]
```diff
@@ -79,10 +78,12 @@
     "@types/rollup-plugin-visualizer": "^4.2.1",
     "@typescript-eslint/eslint-plugin": "^5.59.0",
     "@typescript-eslint/parser": "^5.59.0",
+    "@vitest/coverage-c8": "^0.28.4",
```
[the remark this answers is not in the document]
Comment (sidharthv96) [reply package.json:81]: I didn't add that, it was sorted by pnpm I think.

File: .esbuild/server.ts
Comment (sidharthv96) [reply .esbuild/server.ts:1]: Not sure which type to create.

Did you mean `import type { NextFunction, Request, Response } from 'express';` ?

File: demos/dev/example.html (hunk 1 above)
Comment (sidharthv96) [reply demos/dev/example.html:51]: Done.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (sidharthv96) [reply package.json]: Removed.

File: .esbuild/server.ts
Comment (Yokozuna59) [reply .esbuild/server.ts:1]: I was referring to the variables types in the files, e.g., `mermaidCtx` and `newClient` ...etc.

Comment (sidharthv96) [reply .esbuild/server.ts:1]: Ctxs are already typed from esbuild. 


newClient's type is enforced by the `.push`. I've actually removed the newClient variable.

File: .esbuild/server.ts [hunk 3]
```diff
@@ -1,29 +1,84 @@
...
-  );
-
-  mermaidCtx.watch();
-  mermaidIIFECtx.watch();
-  externalCtx.watch();
-  zenuml.watch();
+  chokidar
+    .watch('**/src/**/*.{js,ts,yaml,json}', {
+      ignoreInitial: true,
+      ignored: [/node_modules/, /dist/, /docs/, /coverage/],
```
Comment (Yokozuna59) [inline .esbuild/server.ts:69]: We don't need to make them '**/node_modules/**'?

**Review by Yokozuna59 — APPROVED** [review]
LGTM.

File: .esbuild/server.ts (hunk 3 above)
Comment (sidharthv96) [reply .esbuild/server.ts:69]: Regex seems to work without the`*`s.

**Pull request merged**